tutorials/shaders/shader_reference/shader_functions.rst

```diff
@@ -0,0 +1,5273 @@
+             Sample_i1_j1(p, base).comps,
+             Sample_i1_j0(p, base).comps,
+             Sample_i0_j0(p, base).comps);
+
+    :param s:
+        the sampler to which the texture from which texels will be retrieved is bound.
+
+    :param p:
+        the texture coordinates at which texture will be sampled.
+
+    :param comps:
+        *optional* the component of the source texture (0 -> x, 1 -> y, 2 -> z, 3 -> w) that will be used to generate the resulting vector. Zero if not specified.
+
+    :return:
+        the gathered texel.
+
+    https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/textureGather.xhtml
+
+.. rst-class:: classref-item-separator
+
+----
+
+
+.. rst-class:: classref-method
+
+vec4 **textureGather** ( samplerCube s, vec3 p [, int comps] )
+
+    Gathers four texels from a texture.
+
+    Returns the value::
+
+        vec4(Sample_i0_j1(p, base).comps,
+             Sample_i1_j1(p, base).comps,
+             Sample_i1_j0(p, base).comps,
+             Sample_i0_j0(p, base).comps);
+
+    :param s:
+        the sampler to which the texture from which texels will be retrieved is bound.
+
+    :param p:
+        the texture coordinates at which texture will be sampled.
+
+    :param comps:
+        *optional* the component of the source texture (0 -> x, 1 -> y, 2 -> z, 3 -> w) that will be used to generate the resulting vector. Zero if not specified.
+
+    :return:
+        the gathered texel.
+
+    https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/textureGather.xhtml
+
+.. rst-class:: classref-item-separator
+
+----
+
+
+
+
+.. _shader_func_dFdx:
+
+.. rst-class:: classref-method
+
+|vec_type| **dFdx** ( |vec_type| p )
+
+    Return the partial derivative of ``p`` with respect to the window x coordinate.
+
+    .. note::
+
+        Available only in the fragment shader
+
+    Returns either `dFdxCoarse<shader_func_dFdxCoarse>` or `dFdxFine<shader_func_dfdxFine>`. The implementation may choose which calculation to perform based upon factors
+    such as performance or the value of the API GL_FRAGMENT_SHADER_DERIVATIVE_HINT hint.
+
+    .. warning::
+        Expressions that imply higher order derivatives such as ``dFdx(dFdx(n))`` have undefined results, as do mixed-order derivatives such as ``dFdx(dFdy(n))``.
+
+    :param p:
+        the expression of which to take the partial derivative.
+
+        .. note::
+
+            It is assumed that the expression ``p`` is continuous and therefore expressions evaluated via non-uniform control flow may be undefined.
+
+    :return:
+        the partial derivative of ``p``.
+
+    https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/dFdx.xhtml
+
+.. rst-class:: classref-item-separator
+
+----
+
+
+
+
+.. _shader_func_dFdxCoarse:
+
+.. rst-class:: classref-method
+
+|vec_type| **dFdxCoarse** ( |vec_type| p )
+
+    Return the partial derivative of ``p`` with respect to the window x coordinate.
+
+    .. note::
+
+        Available only in the fragment shader
```

Comment:
```suggestion
        Available only in the fragment shader
        Not available when using the GL_Compatibility rendering backend.
```

The same note should be added for dFdyCoarse, dFdxFine, dFdyFine, fwidthFine, and fwidthCoarse